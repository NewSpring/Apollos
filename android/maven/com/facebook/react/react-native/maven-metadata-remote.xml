<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>28.0.0</release>
    <versions>
      <version>28.0.0</version>
    </versions>
    <lastUpdated>20180702224301</lastUpdated>
  </versioning>
</metadata>
