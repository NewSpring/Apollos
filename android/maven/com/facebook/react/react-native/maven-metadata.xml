<?xml version="1.0" encoding="UTF-8"?>
<metadata>
  <groupId>com.facebook.react</groupId>
  <artifactId>react-native</artifactId>
  <versioning>
    <release>25.0.0</release>
    <versions>
      <version>25.0.0</version>
    </versions>
    <lastUpdated>20180118183717</lastUpdated>
  </versioning>
</metadata>
