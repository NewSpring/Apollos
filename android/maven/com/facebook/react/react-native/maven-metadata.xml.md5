4d33011c5cd3744e3f0a6d3462b247db