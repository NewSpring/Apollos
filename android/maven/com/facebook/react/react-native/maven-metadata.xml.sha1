8ec2ecca6c881de314f48a8ab786d36bf4a50ada