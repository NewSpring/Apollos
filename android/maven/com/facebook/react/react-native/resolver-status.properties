#NOTE: This is an internal implementation file, its format can be changed without prior notice.
#Thu Jan 18 10:37:17 PST 2018
maven-metadata-remote.xml.lastUpdated=1516300637918
