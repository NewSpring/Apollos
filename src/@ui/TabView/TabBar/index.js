import { TabBar } from 'react-native-tab-view';
import Color from 'color';
import { compose, withProps } from 'recompose';
import styled from '@ui/styled';
import { withTheme } from '@ui/theme';

import Label from './Label';
import Indicator from './Indicator';

const withStyles = compose(
  styled(({ theme }) => ({
    backgroundColor: theme.colors.background.primary,
  }), 'TabBar'),
  withTheme(({ theme, indicatorColor }) => ({
    indicatorColor: indicatorColor || (
      Color(theme.colors.text.primary)
        .mix(Color(theme.colors.background.primary)).rgb().string()
    ),
  })),
);

export default compose(
  withStyles,
  withProps({
    renderLabel: Label,
    renderIndicator: Indicator,
  }),
)(TabBar);
