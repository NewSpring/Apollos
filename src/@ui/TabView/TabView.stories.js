import React from 'react';
import { View, StyleSheet } from 'react-native';
import { storiesOf } from '@storybook/react-native';

import TabView, { SceneMap } from './';

const FirstRoute = () => <View style={[{ flex: 1, backgroundColor: '#ff4081' }]} />;
const SecondRoute = () => <View style={[{ flex: 1, backgroundColor: '#673ab7' }]} />;

storiesOf('TabView', module)
  .add('default', () => (
    <View style={StyleSheet.absoluteFill}>
      <TabView
        routes={[
          { key: 'first', title: 'First' },
          { key: 'second', title: 'Second' },
        ]}
        renderScene={SceneMap({
          first: FirstRoute,
          second: SecondRoute,
        })}
      />
    </View>
  ));
