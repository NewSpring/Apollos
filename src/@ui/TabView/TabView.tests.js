import React from 'react';
import { View } from 'react-native';
import renderer from 'react-test-renderer';
import { TabViewPagerPan } from 'react-native-tab-view';
import Providers from '@ui/TestProviders';

import TabView, { SceneMap } from './';

const FirstRoute = () => <View style={[{ flex: 1, backgroundColor: '#ff4081' }]} />;
const SecondRoute = () => <View style={[{ flex: 1, backgroundColor: '#673ab7' }]} />;

describe('TabView Component', () => {
  it('renders', () => {
    const tree = renderer.create(
      <Providers>
        <TabView
          routes={[
            { key: 'first', title: 'First' },
            { key: 'second', title: 'Second' },
          ]}
          renderScene={SceneMap({
            first: FirstRoute,
            second: SecondRoute,
          })}
          renderPager={props => <TabViewPagerPan {...props} />}
        />
      </Providers>,
    );
    expect(tree).toMatchSnapshot();
  });
});
