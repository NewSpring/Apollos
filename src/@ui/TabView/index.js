import React from 'react';
import PropTypes from 'prop-types';
import { Dimensions, Platform } from 'react-native';
import { TabViewAnimated, SceneMap } from 'react-native-tab-view';
import { compose, withProps, withState, setPropTypes, defaultProps } from 'recompose';
import isFunction from 'lodash/isFunction';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

import TabBar from './TabBar';

const initialLayout = {
  height: 0,
  width: Dimensions.get('window').width,
};

const withStyles = styled({ flex: 1 }, 'TabView');

const defaultHeaderRenderer = ({ barStyle }) => (
  withThemeMixin(({ theme }) => ({
    colors: {
      ...theme.colors,
      text: {
        ...theme.colors.text,
        primary: barStyle === 'dark-content' ? theme.colors.darkPrimary : theme.colors.lightPrimary,
      },
    },
  }))(props => <TabBar {...props} />)
);

const TabView = compose(
  setPropTypes({
    initialIndex: PropTypes.number,
    routes: PropTypes.arrayOf(PropTypes.shape({
      key: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
    })).isRequired,
    renderScene: PropTypes.func.isRequired,
    renderHeader: PropTypes.func,
    swipeEnabled: PropTypes.bool,
  }),
  defaultProps({
    initialIndex: 0,
    swipeEnabled: Platform.OS !== 'web',
  }),
  withStyles,
  withState('index', 'onIndexChange', ({ initialIndex }) => initialIndex),
  withProps((props) => {
    function onIndexChange(index) {
      props.onIndexChange(index);
      if (isFunction(props.onChange)) props.onChange(index);
    }

    return ({
      navigationState: {
        index: props.index,
        routes: props.routes.map(routeProps => ({
          ...routeProps,
          jumpTo(key) {
            const index = props.routes.findIndex(r => (r.key === key));
            return props.onIndexChange(index);
          },
        })),
      },
      initialLayout,
      renderHeader: props.renderHeader ? props.renderHeader : defaultHeaderRenderer(props),
      onIndexChange,
    });
  }),
)(TabViewAnimated);


export { SceneMap, TabBar };
export default TabView;
