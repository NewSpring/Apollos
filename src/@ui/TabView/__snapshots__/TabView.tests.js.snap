// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabView Component renders 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    collapsable={undefined}
    style={
      Object {
        "backgroundColor": "#6bac43",
        "elevation": 4,
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 0.5,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 0.5,
        "zIndex": 1,
      }
    }
  >
    <View
      collapsable={undefined}
      pointerEvents="none"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        collapsable={undefined}
        style={
          Object {
            "backgroundColor": "rgb(181, 214, 161)",
            "bottom": 0,
            "height": 3,
            "left": 0,
            "position": "absolute",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
            "width": 375,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "overflow": "scroll",
        }
      }
    />
  </View>
  <View
    collapsable={undefined}
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 1500,
      }
    }
  >
    <View
      style={
        Object {
          "width": 750,
        }
      }
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#ff4081",
              "flex": 1,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "width": 750,
        }
      }
      testID={undefined}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#673ab7",
              "flex": 1,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
